core/src/main/java/io/micronaut/core/io/service/ServiceScanner.java

```diff
@@ -119,12 +119,28 @@ private Set<String> computeStandardServiceTypeNames(URL url) {
         return typeNames;
     }
 
+    private boolean isWebSphereClassLoader() {
+        return classLoader.getClass().getName().startsWith("com.ibm.ws.classloader");
+    }
+
+    private String buildResourceSearchPath() {
+        String path = "META-INF/micronaut/" + serviceName;
+
+        if (isWebSphereClassLoader()) {
+            // Special case WebSphere classloader
+            // https://github.com/micronaut-projects/micronaut-core/issues/....
```

Comment:
can you add a correct link?

Reply:
Done: [79fbcc8](https://github.com/micronaut-projects/micronaut-core/pull/9906/commits/79fbcc8360288ee3d7487311ba5e09a332ada159)